libpod/sql_state.go

```diff
@@ -453,6 +464,32 @@ func (s *SQLState) UpdateContainer(ctr *Container) error {
 	}
 	newState.FinishedTime = finishedTime
 
+	// Do we need to replace the container's netns?
+	if netNSPath != "" {
+		// Check if the container's old state has a good netns
+		if ctr.state.NetNS != nil && netNSPath == ctr.state.NetNS.Path() {
+			newState.NetNS = ctr.state.NetNS
+		} else {
+			// Tear down the existing namespace
+			if err := s.runtime.teardownNetNS(ctr); err != nil {
```

Comment:
Is there a change that the ctr does not have an existing NS?

Reply:
The NS path would be "" if there was no existing NS